crates/sui-framework/src/natives/test_utils.rs

```diff
@@ -18,3 +19,32 @@ pub fn destroy(
     args.pop_back();
     Ok(NativeResult::ok(legacy_test_cost(), smallvec![]))
 }
+
+pub fn create_one_time_witness(
+    context: &mut NativeContext,
+    mut ty_args: Vec<Type>,
+    args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.len() == 1);
+    debug_assert!(args.is_empty());
+
+    let ty = ty_args.pop().unwrap();
+    let type_tag = context.type_to_type_tag(&ty)?;
+    let type_layout = context.type_to_type_layout(&ty)?;
+
+    let struct_layout = match type_layout {
+        Some(MoveTypeLayout::Struct(l)) => l,
+        _ => return Ok(NativeResult::err(InternalGas::new(1), 0)),
+    };
```

Comment:
This can now be a `let ... else`!

```suggestion
    let Some(MoveTypeLayout::Struct(struct_layout)) = type_layout else {
        return Ok(NativeResult::err(InternalGas::new(1), 0));
    };
```